docker/Dockerfile.build

```diff
@@ -0,0 +1,61 @@
+# syntax = docker/dockerfile:1.4
+FROM bitnami/minideb:latest as main
+
+LABEL org.opencontainers.image.source https://github.com/aurae-runtime/aurae
+
+## Define ARGs
+ARG RUSTUP_TOOLCHAIN=stable
+ARG BUF_VERSION=1.11.0
+ARG VALE_VERSION=2.21.3
+ARG PROTOC_VERSION=1.5.1
+
+## Install packages
+RUN  apt-get update && \
+    install_packages  --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    curl \
+    libclang-dev \
+    libdbus-1-dev \
+    libseccomp-dev \
+    protobuf-compiler \
+    protobuf-compiler \
+    && rm -rf /var/lib/apt/lists/*
+
+## Setup protoc-gen-doc
+RUN curl  -O -J -L https://github.com/pseudomuto/protoc-gen-doc/releases/download/v${PROTOC_VERSION}/protoc-gen-doc_${PROTOC_VERSION}_linux_amd64.tar.gz && \
+    tar -xzf protoc-gen-doc_1.5.1_linux_amd64.tar.gz && \
+    chmod +x protoc-gen-doc && \
+    mv protoc-gen-doc /usr/local/bin/protoc-gen-doc && \
+    rm protoc-gen-doc_1.5.1_linux_amd64.tar.gz
+
+## Setup Rust
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain ${RUSTUP_TOOLCHAIN}
```

Comment:
Are we actually going to be "curling" and installing `rustup` on every pull request? Same goes for the github link above and the others below.

Reply:
No, these will be caches as docker layers so they should only re-run if we need to change the versions we want in the image.